autogen/oai/client.py

```diff
@@ -107,18 +314,23 @@ def __init__(self, *, config_list: Optional[List[Dict[str, Any]]] = None, **base
         openai_config, extra_kwargs = self._separate_openai_config(base_config)
         if type(config_list) is list and len(config_list) == 0:
             logger.warning("openai client was provided with an empty config_list, which may not be intended.")
+
+        self._clients: List[Client] = []
+        self._config_list: List[Dict[str, Any]] = []
+
         if config_list:
             config_list = [config.copy() for config in config_list]  # make a copy before modifying
-            self._clients: List[OpenAI] = [
-                self._client(config, openai_config) for config in config_list
-            ]  # could modify the config
-            self._config_list = [
-                {**extra_kwargs, **{k: v for k, v in config.items() if k not in self.openai_kwargs}}
-                for config in config_list
-            ]
+            for config in config_list:
+                activated = self._register_openai_client(config, openai_config)  # could modify the config
+                self._config_list.append(
+                    {
+                        "config": {**extra_kwargs, **{k: v for k, v in config.items() if k not in self.openai_kwargs}},
+                        "activated": activated,
```

Comment:
Some note here to explain what each field means. Alternatively we can also use a Pydantic model here so we can start to reduce number of dictionaries in the framework internal.

Reply:
removed extra fields and added placeholder class